### PR TITLE
Power and stats vendor implementation

### PR DESCRIPTION
**Pull request by shwethabolumbu**
Power aidl is being moved to
vendor-hardware-interfaces

Using default AIDL for power stats hal as
is from hardware-interfaces

Tracked-On: OAM-126029

### CODE REVIEW

File: power/aidl/apex/Android.bp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    name: "com.android.hardware.power.intel",
+    manifest: "apex_manifest.json",
+    key: "com.android.hardware.key",
+    certificate: ":com.android.hardware.certificate",
+    file_contexts: "file_contexts",
+    updatable: false,
+    vendor: true,
+
+    // Bundle the Power and PowerStats HALs into this one APEX.
+    binaries: [
```
Comment (vilasrk) [inline power/aidl/apex/Android.bp:45]: Binary name is android.hardware.power-service.intel  ,But in source binaries it is mentioned as "android.hardware.power-service.example", "android.hardware.power.stats-service.example",

**Review by vilasrk — CHANGES_REQUESTED** [review]

File: power/aidl/apex/file_contexts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+(/.*)?                                                      u:object_r:vendor_file:s0
+/etc(/.*)?                                                  u:object_r:vendor_configs_file:s0
```
Comment (vilasrk) [inline power/aidl/apex/file_contexts:2]: Selinux permissions are not in line with binary name

**Review by vilasrk — CHANGES_REQUESTED** [review]

File: power/aidl/apex/Android.bp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        "android.hardware.power-service.example",
+        "android.hardware.power.stats-service.example",
+    ],
+    prebuilts: [
```
Comment (vilasrk) [inline power/aidl/apex/Android.bp:49]: xml name might need to change it to power-default.intel.xml?

**Review by vilasrk — CHANGES_REQUESTED** [review]

**Comment by shwethabolumbu** [comment]
Addressed the review comments. 
Moved power and stats into a common apex.

**Review by kaushlen — APPROVED** [review]

**Review by vilasrk — APPROVED** [review]
LGTM

**Pull request merged**